e2e/__tests__/films.test.js

```diff
@@ -0,0 +1,138 @@
+const request = require('../request');
+const db = require('../db');
+const mongoose = require('mongoose');
+
+const actor1 = {
+  name: 'Winona Ryder',
+  dob: 1971,
+  pob: 'Winona, Minnesota'
+};
+
+function postActor(actor) {
+  return request
+    .post('/api/actors')
+    .send(actor)
+    .expect(200)
+    .then(({ body }) => body);
+}
+
+const studio1 = {
+  name: 'Warner Brothers',
+  address: {
+    city: 'Burbank',
+    state: 'California',
+    country: 'USA'
+  }
+};
+
+function postStudio(studio) {
+  return request
+    .post('/api/studios')
+    .send(studio)
+    .expect(200)
+    .then(({ body }) => body);
+}
+
+describe('films api', () => {
+  beforeEach(() => {
+    return db.dropCollection('films');
+  });
+
+  const data = {
+    title: 'Beetlejuice',
+    studio: new mongoose.Types.ObjectId(),
+    released: 1988,
+    cast: [
+      {
+        role: 'Lydia Deetz',
+        actor: new mongoose.Types.ObjectId()
+      }
+    ]
+  };
+
+  function postFilm(film) {
```

Comment:
just for formatting's sake, this can be grouped with your other posting functions at the top